boltdesignsystem/bolt
# DS-407: Update modal demos to use [data-bolt-modal-target] convention

## Jira

https://pegadigitalit.atlassian.net/browse/DS-407

## Summary

Update Modal demos and docs to use the `[data-bolt-modal-target]` convention. Update copy in demos.

## Details

I updated the Modal component demos and instances of Modal in other component demos and "Pages". I also updated some of the copy in the Modal component docs to reflect the new recommended convention.

> Note: fixes a [Listing Teaser dependency bug](https://github.com/boltdesignsystem/bolt/pull/2193/files#diff-dc47f6b3774425192c0192d558ad761b8a5bc1689442d96a3533ffbf643ca20eL20) which was required to get e2e tests to pass.

## How to test

- Review [Modal docs](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=viewall-components-modal). Verify modals still toggle as expected. Review copy changes.
- Review other modified component docs and pages:
    - [Device Viewer](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=components-device-viewer-content-magnifier-modal)
    - [Trigger Component](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=components-trigger-advanced-usage)
    - [Calculator Results](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=pages-d8-results) (click "email my results")
    - [Academy Topic Detail](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=pages-acd-topic-detail-page) (click first article image)
    - [BOC Gallery](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=pages-boc-gallery-phase1) (click "more filters")
    - [BOC Listing](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=pages-boc-listing-pegaworld-phase1) (click "filters")
    
## Release Notes
- The `[on-click]`/`[on-click-target]` method of toggling a modal is deprecated. Instead use `[data-bolt-modal-target]`. See [the docs](https://feature-ds-407-update-modal-docs-demos.boltdesignsystem.com/pattern-lab/?p=components-modal-usage-javascript) for more info (TODO: update link after `v3.6.0` release).
